tools/rpkg/src/statement.cpp

```diff
@@ -67,10 +63,9 @@ SEXP RApi::Prepare(SEXP connsexp, SEXP querysexp) {
 	auto stmtholder = new RStatement();
 	stmtholder->stmt = move(stmt);
 
-	SEXP retlist = r.Protect(NEW_LIST(6));
+	cpp11::list retlist(NEW_LIST(6));
 
-	SEXP stmtsexp = r.Protect(R_MakeExternalPtr(stmtholder, R_NilValue, R_NilValue));
-	R_RegisterCFinalizer(stmtsexp, (void (*)(SEXP))duckdb_finalize_statement_R);
+	cpp11::external_pointer<RStatement> stmtsexp(R_MakeExternalPtr(stmtholder, R_NilValue, R_NilValue));
```

Comment:
Why did you kill the `duckdb_finalize_statement_R()` finalizer? Shouldn't we keep that to ensure proper cleanup?